x-pack/legacy/plugins/apm/public/components/app/Main/route_config/index.tsx

```diff
@@ -107,6 +111,74 @@ export const routes: BreadcrumbRoute[] = [
     ),
     name: RouteName.AGENT_CONFIGURATION
   },
+
+  {
+    exact: true,
+    path: '/settings/agent-configuration/create',
+    breadcrumb: i18n.translate(
+      'xpack.apm.breadcrumb.settings.agentConfigurationTitle',
+      { defaultMessage: 'Create Agent Configuration' }
+    ),
+    name: RouteName.AGENT_CONFIGURATION_CREATE,
+    component: () => {
+      const { search } = history.location;
```

Comment:
Controversy #1: retrieving `search` from `history` singleton. Everywhere else we get it from `useLocation` hook.
The problem with hooks is that they only work inside react components (not vanilla js) so in general I think when given the choice it's better to use the singleton approach (for consistency).

afact there are no difference between the two approaches. LMK if there are!